impeller/renderer/backend/metal/compute_pass_mtl.mm

```diff
@@ -42,231 +47,113 @@
-                 size_t bind_index,
-                 const Sampler& sampler,
-                 const Texture& texture) {
-  if (!sampler.IsValid() || !texture.IsValid()) {
+// |ComputePass|
+bool ComputePassMTL::BindResource(ShaderStage stage,
+                                  DescriptorType type,
+                                  const SampledImageSlot& slot,
+                                  const ShaderMetadata& metadata,
+                                  std::shared_ptr<const Texture> texture,
+                                  std::shared_ptr<const Sampler> sampler) {
+  if (!sampler->IsValid() || !texture->IsValid()) {
     return false;
   }
 
-  pass.SetTexture(bind_index, TextureMTL::Cast(texture).GetMTLTexture());
-  pass.SetSampler(bind_index, SamplerMTL::Cast(sampler).GetMTLSamplerState());
+  pass_bindings_cache_.SetTexture(slot.texture_index,
+                                  TextureMTL::Cast(*texture).GetMTLTexture());
+  pass_bindings_cache_.SetSampler(
+      slot.texture_index, SamplerMTL::Cast(*sampler).GetMTLSamplerState());
   return true;
 }
 
-bool ComputePassMTL::EncodeCommands(const std::shared_ptr<Allocator>& allocator,
-                                    id<MTLComputeCommandEncoder> encoder,
-                                    const ISize& grid_size,
-                                    const ISize& thread_group_size) const {
-  if (grid_size.width == 0 || grid_size.height == 0) {
-    return true;
+fml::Status ComputePassMTL::Compute(const ISize& grid_size) {
+  if (grid_size.IsEmpty()) {
+    return fml::Status(fml::StatusCode::kUnknown,
+                       "Invalid grid size for compute command.");
   }
-
-  ComputePassBindingsCache pass_bindings(encoder);
-
-  fml::closure pop_debug_marker = [encoder]() { [encoder popDebugGroup]; };
-  for (const ComputeCommand& command : commands_) {
-#ifdef IMPELLER_DEBUG
-    fml::ScopedCleanupClosure auto_pop_debug_marker(pop_debug_marker);
-    if (!command.label.empty()) {
-      [encoder pushDebugGroup:@(command.label.c_str())];
-    } else {
-      auto_pop_debug_marker.Release();
-    }
-#endif
-
-    pass_bindings.SetComputePipelineState(
-        ComputePipelineMTL::Cast(*command.pipeline)
-            .GetMTLComputePipelineState());
-
-    for (const BufferAndUniformSlot& buffer : command.bindings.buffers) {
-      if (!Bind(pass_bindings, *allocator, buffer.slot.ext_res_0,
-                buffer.view.resource)) {
-        return false;
-      }
-    }
-
-    for (const TextureAndSampler& data : command.bindings.sampled_images) {
-      if (!Bind(pass_bindings, data.slot.texture_index, *data.sampler,
-                *data.texture.resource)) {
-        return false;
-      }
+  // TODO(dnfield): use feature detection to support non-uniform threadgroup
+  // sizes.
+  // https://github.com/flutter/flutter/issues/110619
+  auto width = grid_size.width;
+  auto height = grid_size.height;
+
+  auto maxTotalThreadsPerThreadgroup = static_cast<int64_t>(
```

Comment:
Here and elsewhere, the use `underscore_case` for vars instead of `camelCase`.

Reply:
Oh, I see this was just code that was moved. Just the same, a drive-by change to make this conform the style would be good.